hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestUpdateTable.scala

```diff
@@ -204,4 +204,48 @@ class TestUpdateTable extends HoodieSparkSqlTestBase {
       }
     })
   }
+
+  test("Test Add Column and Update Table") {
+    withTempDir { tmp =>
```

Comment:
@qidian99 
thanks for your contribution
I ran this UT directly in the main branch， expected to fail but finally succeeded
could you pls check your UT  thanks

Reply:
Thanks for the timely reply. I changed the UT to manually set partition pruning to true.

@stream2000 and I both tested on master branch and the test will fail
![image](https://user-images.githubusercontent.com/20527912/217998452-6f3b0360-905a-479b-8679-6265ab315ac8.png)

Reply:
@qidian99 can you please paste the whole stacktrace? Would like to understand better what exactly is failing

Reply:
I see you pasted the stacktrace failing when you query your data via server. Can you please paste the stacktrace of this particular test failing? 

I want to better understand which operation is failing in this test

Reply:
@qidian99   only non-partitioned tables has this problem？